pkg/lifecycle/daemon/routes_navcycle_completestep.go

```diff
@@ -56,6 +57,7 @@ func (d *NavcycleRoutes) completeStep(c *gin.Context) {
 }
 
 func (d *NavcycleRoutes) handleAsync(errChan chan error, debug log.Logger, step api.Step, stepID string) {
+	defer d.completeMut.Unlock()
```

Comment:
Feels weird to have lock and unlock in different functions. What do you thing about change the `go d.handleAsync` above to a `go func() { ... }` and defer the unlock in there? so they're close by?

Reply:
Makes sense